tests/OmniSharp.Cake.Tests/CompletionFacts.cs

```diff
@@ -0,0 +1,150 @@
+
+        public CompletionFacts(ITestOutputHelper testOutput) : base(testOutput)
+        {
+            _logger = LoggerFactory.CreateLogger<AutoCompleteFacts>();
+        }
+
+        protected override string EndpointName => OmniSharpEndpoints.Completion;
+
+        [Fact]
+        public async Task ShouldGetCompletionFromHostObject()
+        {
+            const string input = @"TaskSe$$";
+
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("CakeProject", shadowCopy : false))
+            using (var host = CreateOmniSharpHost(testProject.Directory))
+            {
+                var fileName = Path.Combine(testProject.Directory, "build.cake");
+                var completions = await FindCompletionsAsync(fileName, input, host);
+
+                Assert.Contains("TaskSetup", completions.Items.Select(c => c.Label));
+                Assert.Contains("TaskSetup", completions.Items.Select(c => c.InsertText));
+            }
+        }
+
+        [Fact]
+        public async Task ShouldGetCompletionFromDSL()
+        {
+            const string input =
+                @"Task(""Test"")
+                    .Does(() => {
+                        Inform$$
+                    });";
+
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("CakeProject", shadowCopy : false))
+            using (var host = CreateOmniSharpHost(testProject.Directory))
+            {
+                var fileName = Path.Combine(testProject.Directory, "build.cake");
+                var completions = await FindCompletionsAsync(fileName, input, host);
+
+                Assert.Contains("Information", completions.Items.Select(c => c.Label));
+                Assert.Contains("Information", completions.Items.Select(c => c.InsertText));
+            }
+        }
+
+        [Fact]
+        public async Task ShouldResolveFromDSL()
+        {
+            const string input =
+                @"Task(""Test"")
+                    .Does(() => {
+                        Inform$$
+                    });";
+
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("CakeProject", shadowCopy : false))
+            using (var host = CreateOmniSharpHost(testProject.Directory))
+            {
+                var fileName = Path.Combine(testProject.Directory, "build.cake");
+                var completion = (await FindCompletionsAsync(fileName, input, host))
+                    .Items.First(x => x.Preselect && x.InsertText == "Information");
+
+                var resolved = await ResolveCompletionAsync(completion, host);
+
+                Assert.StartsWith(
+                    "```csharp\nvoid Information(string format, params object[] args)",
+                    resolved.Item?.Documentation);
+            }
+        }
+
+        // [Fact]
+        // public async Task ShouldGetCompletionWithAdditionalTextEdits()
+        // {
+        //     const string input = @"Regex.Repl$$";
```

Comment:
Anyone know how to get a completion with Additional Text? In Cake, e.g. we don't have `System.Text.RegularExpressions` namespace imported by default. This was my attempt to get a completion from `CompletionItemExtensions.TypeImportCompletionProvider`. I also tried `var regex == new Rege$$`, but that didn't work.

Reply:
it's disabled by default, you need to pass in an option `RoslynExtensionsOptions:EnableImportCompletion`
https://github.com/OmniSharp/omnisharp-roslyn/blob/master/tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs#L1399

Reply:
Ah, thank you!

Reply:
Auch! Does AdditionalTextEdits always contain the whole buffer? //cc @333fred 
![image](https://user-images.githubusercontent.com/7863439/93217725-12163880-f772-11ea-8f4a-4fa6b2e15629.png)

Reply:
Unfortunately, yes.

Reply:
> Unfortunately, yes. 

Thanks for the quick reply. In that case I think I'll just remove all additional texts for Cake files. Can Completion return Additional text, or is it just the Resolver that adds this?

Reply:
Override completion can do this as well, and those are not usually so long.

Reply:
Thanks! I managed to get override completion in at least. Or if it doesn't bring in an unimported type.